light/provider/http/http.go

```diff
@@ -103,8 +103,7 @@ func (p *http) validatorSet(ctx context.Context, height *int64) (*types.Validato
 		for attempt := 1; attempt <= maxRetryAttempts; attempt++ {
 			res, err := p.client.Validators(ctx, height, &page, &maxPerPage)
 			if err != nil {
-				// TODO: standardize errors on the RPC side
-				if regexpMissingHeight.MatchString(err.Error()) {
+				if err = errors.Cause(err); err == ctypes.ErrHeightNotAvailable || err == ctypes.ErrInvalidHeight {
```

Comment:
Would `errors.Is(err, ctypes.ErrHeightNotAvailable) || ...` be preferable? At least we don't overshadow the original `err`.